clang/include/clang/Support/RISCVVIntrinsicUtils.h

```diff
@@ -416,8 +416,10 @@ class RVVIntrinsic {
   RVVTypePtr getOutputType() const { return OutputType; }
   const RVVTypes &getInputTypes() const { return InputTypes; }
   llvm::StringRef getBuiltinName() const { return BuiltinName; }
-  llvm::StringRef getName() const { return Name; }
-  llvm::StringRef getOverloadedName() const { return OverloadedName; }
+  llvm::StringRef getName() const { return "__riscv_" + Name; }
```

Comment:
How can you return a StringRef of a concentation? Who owns the concatenated string? StringRef doesn't take ownership.

Reply:
`Name` is a std::string. The addition creates a new temporary std::string. That only has a lifetime of the body of the function. The StringRef points to the temporary std::string.

Reply:
Isn't it returning *this?
https://cplusplus.com/reference/string/string/operator+=/

Reply:
I mean, return the temporary string's raw data pointer(const char*).

Reply:
You can make the function return a std::string.

Reply:
That raw pointer is allocated on the heap. It has the same lifetime as the std::string temporary.

Reply:
I just can't find where is `getName` and `getOverloadedName`, I just comment out this 2 member functions and run `ninja check-clang`, it passed.
Can we delete them?

Reply:
Sure